src/wait.rs

```diff
@@ -118,7 +117,7 @@ impl SnapshotWaiter {
                 for snapshot in snapshots {
                     if let Some(ref found_id) = snapshot.snapshot_id {
                         if let Some(ref found_state) = snapshot.state {
-                            if snapshot_id.as_ref() == found_id && &state == found_state {
+                            if snapshot_id.as_ref() == found_id && state == found_state.as_str() {
```

Comment:
If you change this to `if let Some(found_state)` then you can avoid the dereference change below - seems like we don't actually want a ref now.